src/app/common/components/AddEditTokenModal/AddEditTokenForm.tsx

```diff
@@ -0,0 +1,256 @@
+    const loginUrl = `${clusterLoginUrl}?${searchParams.toString()}`;
+    const loginWindow = window.open(loginUrl, '_blank');
+    loginWindow.focus();
+    loginWindowRef.current = loginWindow;
+  };
+
+  const handleStorageChanged = () => {
+    if (loginAttemptIdRef.current && loginWindowRef.current) {
+      const authCode = window.localStorage.getItem(`oauth-code-${loginAttemptIdRef.current}`);
+      if (authCode) {
+        setOAuthResponseCode(authCode);
+        loginWindowRef.current.close();
+        window.localStorage.removeItem(`oauth-code-${loginAttemptIdRef.current}`);
+        loginAttemptIdRef.current = null;
+      }
+    }
+  };
+
+  useEffect(() => {
+    window.addEventListener('storage', handleStorageChanged);
+    return () => window.removeEventListener('storage', handleStorageChanged);
+  }, []);
+
+  return (
+    <Form onSubmit={handleSubmit}>
+      <FormGroup
+        label="Name"
+        isRequired
+        fieldId={FieldKey.name}
+        helperTextInvalid={touched.name && errors.name}
+        isValid={!(touched.name && errors.name)}
+      >
+        <TextInput
+          onChange={formikHandleChange}
+          onInput={formikSetFieldTouched(FieldKey.name)}
+          onBlur={handleBlur}
+          value={values.name}
+          name={FieldKey.name}
+          type="text"
+          id={FieldKey.name}
+          isDisabled={addEditStatus.mode === AddEditMode.Edit}
+          isValid={!(touched.name && errors.name)}
+        />
+      </FormGroup>
+      <FormGroup
+        label="Associate with cluster"
+        isRequired
+        fieldId={FieldKey.associatedClusterName}
+        helperTextInvalid={touched.associatedClusterName && errors.associatedClusterName}
+        isValid={!(touched.associatedClusterName && errors.associatedClusterName)}
+      >
+        <SimpleSelect
+          id={FieldKey.associatedClusterName}
+          onChange={(selection) => setFieldValue(FieldKey.associatedClusterName, selection)}
+          options={['foo', 'bar', 'baz']} // NATODO
+          value={values.associatedClusterName}
+          placeholderText="Select cluster..."
+        />
+      </FormGroup>
+      <FormGroup
+        label="Token type"
+        isRequired
+        fieldId={FieldKey.tokenType}
+        helperTextInvalid={touched.tokenType && errors.tokenType}
+        isValid={!(touched.tokenType && errors.tokenType)}
+      >
+        <Radio
+          id={`${FieldKey.tokenType}-${TokenType.oAuth}`}
+          name={FieldKey.tokenType}
+          isChecked={values.tokenType === TokenType.oAuth}
+          onChange={(checked) => {
+            if (checked) setFieldValue(FieldKey.tokenType, TokenType.oAuth);
+          }}
+          label="OAuth"
+        />
+        <div className={`${spacing.mtMd} ${spacing.mbLg} ${spacing.mlLg}`}>
+          <TextContent>
+            <Text component="p">
+              Log in to the cluster to automatically generate the token, or enter the token string
```

Comment:
We aren't going to accept user provided OAuth tokens, so I think the copy after the comma should be removed. @vconzola thoughts on that?

Reply:
Not going to hold up this PR on this since this is all under construction.

Reply:
Maybe that's referring to the Service Account field below, and this help text should be above the radio button choice (if it describes both)?

Reply:
@mturley Nope, @eriknelson is correct. That text is leftover from when I thought user could enter an Oauth token string. It's removed from my latest mockups.

Reply:
Ah, ok thanks @vconzola. When we get this all connected we'll have to do a once-over of your mockups to make sure we got all these little details.

Reply:
Are these still your latest mockups? I still see this text in there: https://docs.google.com/presentation/d/1JU_Lw1A-gUO4gBRofZzi74aONcknYRHbqBl_xHxikYQ/edit#slide=id.g801d08d2ef_0_16

Reply:
@mturley Dang! I must have updated the mockup in Balsamiq but never copied it into Slides. Will do that this morning.

Reply:
@mturley Mockups updated with new text. https://docs.google.com/presentation/d/1JU_Lw1A-gUO4gBRofZzi74aONcknYRHbqBl_xHxikYQ/edit?usp=sharing